File: app/src/main/java/com/example/toutiao/activity/MainActivity.java
package com.example.toutiao.activity;

import android.os.Build;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.toutiao.Contact.BaseContact;
import com.example.toutiao.R;
import com.example.toutiao.Base.BaseActivity;
import com.example.toutiao.fragment.HomeFragment;
import com.example.toutiao.fragment.VideoFragment;
import com.example.toutiao.fragment.ProfileFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends BaseActivity<BaseContact.IPresenter> implements BaseContact.IView {

    @BindView(R.id.bottom_nav)
    BottomNavigationView bottomNavigationView;

    @Override
    protected void initViews() {
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
    }

    @Override
    protected void initLayout(@Nullable Bundle savedInstanceState) {
        if (Build.VERSION.SDK_INT >= 21) {
            Window window = getWindow();
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.setStatusBarColor(getResources().getColor(R.color.colorTintBlue));
        }

        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.activity_main_fragment_container, new HomeFragment())
                .commit();

        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                switch (menuItem.getItemId()) {
                    case R.id.item_home:
                        getSupportFragmentManager()
                                .beginTransaction()
                                .replace(R.id.activity_main_fragment_container, new HomeFragment())
                                .commit();
                        return true;
                    case R.id.item_video:
                        getSupportFragmentManager()
                                .beginTransaction()
                                .replace(R.id.activity_main_fragment_container, new VideoFragment())
                                .commit();
                        return true;
                    case R.id.item_profile:
                        getSupportFragmentManager()
                                .beginTransaction()
                                .replace(R.id.activity_main_fragment_container, new ProfileFragment())
                                .commit();
                        return true;
                }
                return false;
            }
        });
    }

    @Override
    protected void initData() {

    }

    @Override
    protected BaseContact.IPresenter setPresenter() {
        return null;
    }
}

File: app/src/main/java/com/example/toutiao/activity/MainActivity.kt
package com.example.toutiao.activity

import android.os.Bundle
import android.widget.Toast
import com.google.android.material.bottomnavigation.BottomNavigationView
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.example.toutiao.R
import com.example.toutiao.base.BaseActivity
import com.example.toutiao.utils.toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    private lateinit var toast: Toast

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        toast = "再按一次退出应用".toast(this)
        val navView: BottomNavigationView = nav_view
        val navController = findNavController(R.id.nav_host_fragment)
        val appBarConfiguration = AppBarConfiguration(setOf(
                R.id.navigation_home, R.id.navigation_dashboard, R.id.navigation_notifications))
        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)
    }

    override fun onBackPressed() {
        if (toast.view.parent == null) {
            toast.show()
        } else {
            finish()
        }
    }
}
